File: Algorithm/Search/1-sherlock-and-array.cpp
#include <iostream>
#include <cstdio>

using namespace std;

int main(int argc, char const *argv[])
{
    int T;
    cin >> T;
    while (T --) {
        int N;
        cin >> N;
        int A[N];
        int sum = 0;
        for (int i = 0; i < N; i ++) {
            scanf("%d", &A[i]);
            sum += A[i];
        }

        int right = sum;
        bool exists = false;

        for (int i = 0; i < N; i ++) {
            int left = sum - right;
            right -= A[i];

            if (left == right) {
                exists = true;
                break;
            }
        }

        if (exists) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }

    return 0;
}

File: Algorithm/Search/1-sherlock-and-array.php
<?php

$handle = fopen("php://stdin", "r");
fscanf($handle, "%d", $t);

while ($t --) {
    fscanf($handle, "%d", $n);
    $arr = array_map('intval', explode(' ', trim(fgets($handle))));

    if ($n == 1) {
        echo "YES\n";
        continue;
    }

    $exists = false;

    $sum = array_sum($arr);
    $right = $sum;

    for ($i=0; $i < $n; $i++) {
        $left = $sum - $right;
        $right = $right - $arr[$i];

        if ($left == $right) {
            $exists = true;
            break;
        }
    }

    echo $exists ? "YES\n" : "NO\n";
}
